Bottom: 41cfabece67c7ccb12c4ada69768bb51537a9053
Top:    12b0ace3981d06c1a29d65dfc8877bb062282b4f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-25 18:14:53 +0100

draw.c: Split "Gathering" routines from real drawing routines (Text)

Moves code from DrawRegularText() into its only caller text_callback()


---

diff --git a/src/draw.c b/src/draw.c
index beb607c..e3c79e3 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -89,7 +89,6 @@ static const BoxType *clip_box = NULL;
 static void DrawEverything (BoxTypePtr);
 static void DrawPPV (int group, const BoxType *);
 static int DrawLayerGroup (int, const BoxType *);
-static void DrawRegularText (LayerTypePtr, TextTypePtr);
 static void AddPart (void *);
 static void SetPVColor (PinTypePtr, int);
 static void DrawEMark (ElementTypePtr, LocationType, LocationType, bool);
@@ -1037,7 +1036,20 @@ DrawRats (BoxTypePtr drawn_area)
 static int
 text_callback (const BoxType * b, void *cl)
 {
-  DrawRegularText ((LayerTypePtr) cl, (TextTypePtr) b);
+  LayerType *layer = cl;
+  TextType *text = (TextType *)b;
+  int min_silk_line;
+
+  if (TEST_FLAG (SELECTEDFLAG, text))
+    gui->set_color (Output.fgGC, layer->SelectedColor);
+  else
+    gui->set_color (Output.fgGC, layer->Color);
+  if (layer == &PCB->Data->SILKLAYER ||
+      layer == &PCB->Data->BACKSILKLAYER)
+    min_silk_line = PCB->minSlk;
+  else
+    min_silk_line = PCB->minWid;
+  DrawTextLowLevel (text, min_silk_line);
   return 1;
 }
 
@@ -1188,12 +1200,6 @@ DrawTextLowLevel (TextTypePtr Text, int min_line_width)
   Cardinal n;
   FontTypePtr font = &PCB->Font;
 
-  if (Gathering)
-    {
-      AddPart (Text);
-      return;
-    }
-
   while (string && *string)
     {
       /* draw lines if symbol is valid and data is present */
@@ -1394,40 +1400,12 @@ DrawArc (LayerTypePtr Layer, ArcTypePtr Arc)
 void
 DrawText (LayerTypePtr Layer, TextTypePtr Text)
 {
-  int min_silk_line;
-  if (!Layer->On)
-    return;
-  if (TEST_FLAG (SELECTEDFLAG, Text))
-    gui->set_color (Output.fgGC, Layer->SelectedColor);
-  else
-    gui->set_color (Output.fgGC, Layer->Color);
-  if (Layer == & PCB->Data->SILKLAYER
-      || Layer == & PCB->Data->BACKSILKLAYER)
-    min_silk_line = PCB->minSlk;
-  else
-    min_silk_line = PCB->minWid;
-  DrawTextLowLevel (Text, min_silk_line);
-}
+  assert (Gathering);
 
-/* ---------------------------------------------------------------------------
- * draws text on a layer
- */
-static void
-DrawRegularText (LayerTypePtr Layer, TextTypePtr Text)
-{
-  int min_silk_line;
-  if (TEST_FLAG (SELECTEDFLAG, Text))
-    gui->set_color (Output.fgGC, Layer->SelectedColor);
-  else
-    gui->set_color (Output.fgGC, Layer->Color);
-  if (Layer == & PCB->Data->SILKLAYER
-      || Layer == & PCB->Data->BACKSILKLAYER)
-    min_silk_line = PCB->minSlk;
-  else
-    min_silk_line = PCB->minWid;
-  DrawTextLowLevel (Text, min_silk_line);
+  AddPart (Text);
 }
 
+
 /* ---------------------------------------------------------------------------
  * draws a polygon on a layer
  */
@@ -1490,7 +1468,7 @@ DrawElementName (ElementTypePtr Element)
 
   if (TEST_FLAG (HIDENAMEFLAG, Element))
     return;
-  DrawTextLowLevel (&ELEMENT_TEXT (PCB, Element), PCB->minSlk);
+  DrawText (NULL, &ELEMENT_TEXT (PCB, Element));
 }
 
 /* ---------------------------------------------------------------------------
@@ -1658,13 +1636,9 @@ EraseArc (ArcTypePtr Arc)
 void
 EraseText (LayerTypePtr Layer, TextTypePtr Text)
 {
-  int min_silk_line;
-  if (Layer == & PCB->Data->SILKLAYER
-      || Layer == & PCB->Data->BACKSILKLAYER)
-    min_silk_line = PCB->minSlk;
-  else
-    min_silk_line = PCB->minWid;
-  DrawTextLowLevel (Text, min_silk_line);
+  assert (Gathering);
+
+  AddPart (Text);
 }
 
 /* ---------------------------------------------------------------------------
@@ -1694,8 +1668,7 @@ EraseElement (ElementTypePtr Element)
     EraseArc (arc);
   }
   END_LOOP;
-  if (!TEST_FLAG (HIDENAMEFLAG, Element))
-    DrawTextLowLevel (&ELEMENT_TEXT (PCB, Element), PCB->minSlk);
+  EraseElementName (Element);
   EraseElementPinsAndPads (Element);
 }
 
@@ -1729,7 +1702,7 @@ EraseElementName (ElementTypePtr Element)
 
   if (TEST_FLAG (HIDENAMEFLAG, Element))
     return;
-  DrawTextLowLevel (&ELEMENT_TEXT (PCB, Element), PCB->minSlk);
+  DrawText (NULL, &ELEMENT_TEXT (PCB, Element));
 }
